sympy/matrices/matrices.py

```diff
@@ -3986,3 +3986,117 @@ def classof(A, B):
 def a2idx(j, n=None):
     from sympy.matrices.common import a2idx as a2idx_
     return a2idx_(j, n)
+
+
+def randMatrixBigDiag(rows, cols=None, lo=0, hi=100, symmetric=False):
+    from sympy.matrices.dense import diag
+    """Return a matrix filled with random values such that no
+    value is greater than the values on the diagonals in the same
```

Comment:
Imports should not be before docstrings. The docstring will not be saved in `__doc__` if you do that.

I'm not sure that this function should be added as it seems obscure.

Reply:
I shifted imports after docstring.
Should I remove `randMatrixBigDiag` methods? @oscarbenjamin